### DISCUSSION TITLE
"Learning CFEngine" book

### DISCUSSION THREAD

**Discussion by zzamboni** [cfengine/core#4516] [Show and tell]
I wrote and (sometimes) maintain the book "Learning CFEngine". The book was published originally by O'Reilly Media in 2012 as _Learning CFEngine 3_, and you can still find copies of it in some bookstores, and online at O'Reilly Learning (formerly Safari Books Online): https://learning.oreilly.com/library/view/learning-cfengine-3/9781449334536/

In 2017 I started self-publishing the book, you can find the latest version at https://leanpub.com/learning-cfengine. You can find additional information (including the source code files from the book) at its website https://cf-learn.info/

The book currently covers CFEngine 3.12, so it's not fully up to date, but it's still a great reference if you want to get started with CFEngine.

Check it out!

**Comment by olehermanse** [comment]
Definitely a great resource. I have my hard copy in front of me, on my (home) office desk. Can recommend to anyone learning CFEngine policy language :) Thanks for sharing!

**Reply by nickanderson** [reply]
Note: I don't think your hard copy covers 3.12.x, for that you need to re-purchase via Leanpub (and I don't know that a hard copy is an option).

**Reply by zzamboni** [reply]
BTW - thanks @olehermanse for prompting me to write this post!

**Comment by nickanderson** [comment]
Indeed, it's a great resource, highly recommended.

**Comment by zzamboni** [comment]
Indeed, the hard copy from O'Reilly covers up to 3.5 IIRC. So far the
Leanpub version is not available on hard copy, but it might be possible if
there's enough interest 😊

**Comment by zzamboni** [comment]
As an additional piece of trivia (for those who are into that sort of thing), the book is written using [Emacs](https://www.gnu.org/software/emacs/) in [Org mode](https://orgmode.org/) format, from which the markup and structure used by Leanpub is exported using the [ox-leanpub](https://github.com/zzamboni/ox-leanpub) package.

\<shameless plug\>
I also wrote a book about the writing/publishing tools and process I use: https://leanpub.com/emacs-org-leanpub
\</shameless plug\>
